From 1b87092e363397010beb55220b6953d129aa9c3b Mon Sep 17 00:00:00 2001
From: Nathan Chancellor <natechancellor@gmail.com>
Date: Wed, 3 Oct 2018 14:28:57 -0700
Subject: [PATCH 06/13] DO-NOT-UPSTREAM: Kbuild: Hide Clang's -Wempty-body
 behind W=1

This doesn't happen often but the few times that it has, it's just
noise. I believe the intention of the warning is to avoid situations
like this:

if (condition);
        statement;

where the user really intended

if (condition)
        statement;

However, GCC's -Wmisleading-indention should have caught any issues of
this form and the remaining warnings are just conditional statements
being used as shortcuts. For example:

drivers/atm/eni.c:244:48: warning: for loop has empty body
[-Wempty-body]
        for (order = 0; (1 << order) < *size; order++);
                                                      ^

Which is just used to get order to the correct size. Another one:

drivers/atm/zatm.c:513:7: warning: while loop has empty body
[-Wempty-body]
        zwait;
             ^

Which is just used to wait while busy. Moving the semi-colon to another
line defeats the purpose of the compact nature of these constructs so
just hide the warning behind W=1 so its use can still be audited but it
won't polute the build.

Link: https://github.com/ClangBuiltLinux/linux/issues/42
Link: https://github.com/ClangBuiltLinux/linux/issues/66
Signed-off-by: Nathan Chancellor <natechancellor@gmail.com>
---
 scripts/Makefile.extrawarn | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/scripts/Makefile.extrawarn b/scripts/Makefile.extrawarn
index 24b2fb1d1297..6003e69adff9 100644
--- a/scripts/Makefile.extrawarn
+++ b/scripts/Makefile.extrawarn
@@ -11,6 +11,7 @@
 # are not supported by all versions of the compiler
 # ==========================================================================
 
+KBUILD_CFLAGS += $(call cc-disable-warning, empty-body)
 KBUILD_CFLAGS += $(call cc-disable-warning, packed-not-aligned)
 
 ifeq ("$(origin W)", "command line")
@@ -31,6 +32,7 @@ warning-1 += $(call cc-option, -Wunused-const-variable)
 warning-1 += $(call cc-option, -Wpacked-not-aligned)
 warning-1 += $(call cc-disable-warning, missing-field-initializers)
 warning-1 += $(call cc-disable-warning, sign-compare)
+warning-1 += $(call cc-option, -Wempty-body)
 
 warning-2 := -Waggregate-return
 warning-2 += -Wcast-align
-- 
2.19.0

